docs/sw2sphinxopenapi.py

```diff
@@ -0,0 +1,142 @@
+"""Generate Markdown documents from an OpenAPI swagger file."""
+
+from __future__ import print_function
+
+import contextlib
+import os
+import os.path
+import re
+import sys
+import textwrap
+
+import yaml
+
+
+class OutputFiles(object):
+    """A context manager to manage a series of files.
+
+    Use like this::
+
+        with OutputFiles() as outfiles:
+            ...
+            if some_condition():
+                f = outfiles.open("filename.txt", "w")
+
+    Each open will close the previously opened file, and the end of the with
+    statement will close the last one.
+
+    """
+    def __init__(self):
+        self.file = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        if self.file:
+            self.file.close()
+        return False
+
+    def open(self, *args, **kwargs):
+        if self.file:
+            self.file.close()
+        self.file = open(*args, **kwargs)
+        return self.file
+
+
+@contextlib.contextmanager
+def redirect_output(file):
```

Comment:
(Again, sorry to read this only now)

This makes it difficult to debug a piece of code with `print` statements, as they will redirect output to the file. I understand what you did, and the resulting code is elegant, but I'm generally averse to modifying the static context.

Reply:
I've dealt with the prints differently in the latest code.